configure error with IBM Spectrum MPI

I am encountering a `configure` error when attempting to compile NetCDF 4.7.3 with IBM Spectrum MPI. 

**configure command**

```console
./configure --prefix=$TROOT_DIR CC=mpicc CXX=mpic++ FC=mpifort CFLAGS="-I${TROOT_DIR}/include -O3" LDFLAGS="-L$TROOT_DIR/lib" --enable-pnetcdf --enable-parallel-tests --enable-netcdf-4 --disable-shared --disable-fsync --disable-cdmremote --disable-dap --disable-doxygen --disable-v2
```
**Error**
```
configure: error: --enable-pnetcdf requires version 1.6.0 or later
```

```
configure:18476: mpicc -c -I/gpfs/alpine/cfd116/scratch/sayerhs/netcdf-test/inst
all//include -O3  conftest.c >&5
In file included from /gpfs/alpine/cfd116/scratch/sayerhs/netcdf-test/install//include/pnetcdf.h:13:0,
                 from conftest.c:127:
/autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-7.4.0/spectrum-mpi-10.3.1.2-20200121-eu2wh5ypz6ipvokwbng2m5ihug6doysh/include/mpi.h: In function 'main':
/autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-7.4.0/spectrum-mpi-10.3.1.2-20200121-eu2wh5ypz6ipvokwbng2m5ihug6doysh/include/mpi.h:280:34: error: static declaration of 'smpi_built_against_version' follows non-static declaration
       OMPI_DECLSPEC const char * smpi_built_against_version(void) { return "SMPI_BUILT_AGAINST_VERSION=" SMPI_VERSION_STR; }
                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~
/autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-7.4.0/spectrum-mpi-10.3.1.2-20200121-eu2wh5ypz6ipvokwbng2m5ihug6doysh/include/mpi.h:279:34: note: previous declaration of 'smpi_built_against_version' was here
       OMPI_DECLSPEC const char * smpi_built_against_version(void);
                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~
```

I am able to fix this by changing `configure.ac`

```diff
--- configure.ac.orig	2019-11-20 18:59:25.000000000 -0500
+++ configure.ac	2020-06-03 12:50:20.543484000 -0400
@@ -1147,8 +1147,7 @@
   # Pnetcdf did not support utf-8 until 1.6.0

   AC_MSG_CHECKING([Is libpnetcdf version 1.6.0 or later?])
-  AC_COMPILE_IFELSE([AC_LANG_PROGRAM([], [[
-#include <pnetcdf.h>
+  AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <pnetcdf.h>]], [[
 #if (PNETCDF_VERSION_MAJOR*1000 + PNETCDF_VERSION_MINOR < 1006)
       choke me
 #endif
@@ -1206,8 +1205,7 @@
    UD_CHECK_HEADER_PATH([pnetcdf.h])

    AC_MSG_CHECKING([if erange-fill is enabled in PnetCDF])
-   AC_COMPILE_IFELSE([AC_LANG_PROGRAM([], [[
-#include <pnetcdf.h>
+   AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <pnetcdf.h>]], [[
 #if !defined(PNETCDF_ERANGE_FILL) || PNETCDF_ERANGE_FILL == 0
       choke me
 #endif]])], [enable_erange_fill_pnetcdf=yes], [enable_erange_fill_pnetcdf=no])
@@ -1223,8 +1221,7 @@
    fi

    AC_MSG_CHECKING([if relax-coord-bound is enabled in PnetCDF])
-   AC_COMPILE_IFELSE([AC_LANG_PROGRAM([], [[
-#include <pnetcdf.h>
+   AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <pnetcdf.h>]], [[
 #if !defined(PNETCDF_RELAX_COORD_BOUND) || PNETCDF_RELAX_COORD_BOUND == 0
       choke me
 #endif]])], [relax_coord_bound_pnetcdf=yes], [relax_coord_bound_pnetcdf=no])
```

**System information**

- Operating System: RHEL 7.6 ppc64le
- HDF5 version: 1.10.1
- Parallel NetCDF info

```
This PnetCDF 1.11.0 was built with the following features:

  --has-c++                   -> yes
  --has-fortran               -> no
  --netcdf4                   -> disabled
  --relax-coord-bound         -> enabled
  --in-place-swap             -> auto
  --erange-fill               -> enabled
  --subfiling                 -> disabled
  --large-single-req          -> disabled
  --null-byte-header-padding  -> disabled
  --burst-buffering           -> disabled
  --profiling                 -> disabled
  --thread-safe               -> disabled
  --debug                     -> disabled

Additional information:

  --version       -> PnetCDF 1.11.0
  --release-date  -> 19 Dec 2018
  --config-date   -> Wed Jun  3 12:16:33 EDT 2020
```

--- a/configure.ac
+++ b/configure.ac
@@ -1177,8 +1177,7 @@ if test "x$enable_pnetcdf" = xyes; then
   # Pnetcdf did not support utf-8 until 1.6.0
 
   AC_MSG_CHECKING([Is libpnetcdf version 1.6.0 or later?])
-  AC_COMPILE_IFELSE([AC_LANG_PROGRAM([], [[
-#include <pnetcdf.h>
+  AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <pnetcdf.h>]], [[
 #if (PNETCDF_VERSION_MAJOR*1000 + PNETCDF_VERSION_MINOR < 1006)
       choke me
 #endif
@@ -1229,8 +1228,7 @@ if test "x$enable_pnetcdf" = xyes; then
    UD_CHECK_HEADER_PATH([pnetcdf.h])
 
    AC_MSG_CHECKING([if erange-fill is enabled in PnetCDF])
-   AC_COMPILE_IFELSE([AC_LANG_PROGRAM([], [[
-#include <pnetcdf.h>
+   AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <pnetcdf.h>]], [[
 #if !defined(PNETCDF_ERANGE_FILL) || PNETCDF_ERANGE_FILL == 0
       choke me
 #endif]])], [enable_erange_fill_pnetcdf=yes], [enable_erange_fill_pnetcdf=no])
@@ -1246,8 +1244,7 @@ if test "x$enable_pnetcdf" = xyes; then
    fi
 
    AC_MSG_CHECKING([if relax-coord-bound is enabled in PnetCDF])
-   AC_COMPILE_IFELSE([AC_LANG_PROGRAM([], [[
-#include <pnetcdf.h>
+   AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <pnetcdf.h>]], [[
 #if !defined(PNETCDF_RELAX_COORD_BOUND) || PNETCDF_RELAX_COORD_BOUND == 0
       choke me
 #endif]])], [relax_coord_bound_pnetcdf=yes], [relax_coord_bound_pnetcdf=no])

